includes/menu.inc

```diff
@@ -1560,7 +1561,9 @@ function menu_set_active_trail($new_trail = NULL) {
     }
 
     $tree = menu_tree_page_data(menu_get_active_menu_name());
-    list($key, $curr) = each($tree);
+
+    $curr = current($tree);
+    next($tree);
```

Comment:
No, I don't think so - both variables ($curr and $tree) are used in the while clause after this code.

Reply:
The code is indeed correct as-is. This is the literal conversion of what each() does: _first_ assign the current value, _then_ advance $tree. This means "$curr = next($tree);" would not be correct, because the while($curr) loop wants to process the _current_ value, which $tree has already advanced beyond.

To say this differently: the while($curr) loop is adjusted to run correctly with this somewhat strange code, and you _can_ make this one line, but that _also_ means you have to adjust the while loop. The one line would be:
```
$curr = current($tree);
``` 
...and then you would have to add a "} else {" to the while loop, as in:
```
    $tree = $curr['below'] ? $curr['below'] : array(); // <<<<<< This code exists; see below
  }
  else {
    next($tree);
  }
  $curr = current($tree);   //// <<<< now the next() is not necessary here
```

**My suggestion:** Either do what is described in this comment, or put a comment above the above next() statement, because it's indeed hard to decipher this:
```
// The while() loop assumes that the $tree array pointer advances already.
```